### PR TITLE
[logstash] Add support to use pattern files

### PR DESCRIPTION
**Pull request by tuananhnguyen-ct**
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

Fixes https://github.com/elastic/helm-charts/issues/685, as we don't want to manage the pattern via a configmap externally.

To use the pattern, add `patterns_dir => "/usr/share/logstash/patterns"` to your filter.

### CODE REVIEW

**Comment by tuananhnguyen-ct** [comment]
@jmlrt can I have a review here please? Thank you.

**Comment by jmlrt** [comment]
Hi @tuananhnguyen-ct,
Thanks for opening this PR.
We'll try to review it soon. Please be patient.

**Comment by jmlrt** [comment]
LGTM⛴

**Comment by jmlrt** [comment]
jenkins test this please

**Review by ebuildy — DISMISSED** [review]

**Comment by tuananhnguyen-ct** [comment]
@jmlrt please review, the test has passed. Thanks a lot.

File: logstash/README.md [hunk 1]
```diff
@@ -121,6 +121,8 @@ useful for the [http input plugin][], for instance.
 | `logstashConfig`          | Allows you to add any config files in `/usr/share/logstash/config/` such as `logstash.yml` and `log4j2.properties` See [values.yaml][] for an example of the formatting                                                              | `{}`                                  |
 | `logstashJavaOpts`        | Java options for Logstash. This is where you should configure the JVM heap size                                                                                                                                                      | `-Xmx1g -Xms1g`                       |
 | `logstashPipeline`        | Allows you to add any pipeline files in `/usr/share/logstash/pipeline/`                                                                                                                                                              | `{}`                                  |
+| `logstashPatternDir`      | Allows you to define a custom directory to store patten files | `/usr/share/logstash/patterns/` |
+| `logstashPipeline`        | Allows you to add any pattern files in `logstashPatternDir` | `{}` |
```
Comment (jmlrt) [inline logstash/README.md:125]:
```suggestion
| `logstashPattern`        | Allows you to add any pattern files in `logstashPatternDir` | `{}` |
```

**Review by jmlrt — CHANGES_REQUESTED** [review]
Hi @tuananhnguyen-ct, this looks mostly good outside of the copy/paste error in the README.
I'll also try to get someone else with better knowledge of Logstash pattern files than me.

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Pull request merged**